sql/moz-fx-data-shared-prod/monitoring/average_ping_sizes_v1/query.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+"""Determine stable table average ping sizes."""
+
+from argparse import ArgumentParser
+from fnmatch import fnmatchcase
+from google.cloud import bigquery
+from functools import partial
+from multiprocessing.pool import ThreadPool
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument("--date", required=True)  # expect string with format yyyy-mm-dd
+parser.add_argument("--project", default="moz-fx-data-shared-prod")
+parser.add_argument("--dataset", default="*_stable")  # pattern
+parser.add_argument("--destination_dataset", default="monitoring")
+parser.add_argument("--destination_table", default="average_ping_sizes_v1")
+
+
+def get_tables(client, project, dataset):
+    """Returns list of all available tables."""
+    sql = f"SELECT dataset_id, table_id FROM `{project}.{dataset}.__TABLES__`"
+
+    try:
+        result = client.query(sql).result()
+        return [(row.dataset_id, row.table_id) for row in result]
+    except Exception as e:
+        print(f"Error querying dataset {dataset}: {e}")
+
+    return []
+
+
+def get_average_ping_size_json(client, date, table):
+    """Returns the average ping sizes for a specific date."""
+    print(table)
+    try:
+        dataset_id = table[0]
+        table_id = table[1]
+
+        sql = f"""
+            WITH total_pings AS (
+                SELECT COUNT(*) as total
+                FROM {dataset_id}.{table_id}
+                WHERE DATE(submission_timestamp) = '{date}'
+            ) SELECT
+                COALESCE(SAFE_DIVIDE(byte_size, total)) AS average_ping_size
+            FROM total_pings, `moz-fx-data-shared-prod.monitoring.stable_table_sizes_v1`
+            WHERE submission_date = '{date}' AND dataset_id = '{dataset_id}' AND table_id = '{table_id}'
+        """
+
+        result = client.query(sql).result()
+        average_ping_size = [row.average_ping_size for row in result][0]
+
+        return {
+            "submission_date": date,
+            "dataset_id": dataset_id,
+            "table_id": table_id,
+            "average_byte_size": average_ping_size,
```

Comment:
We could maintain `byte_size` and `row_count` columns in this table as well. It may be useful to be able to define a dashboard that correlates these 3 values, so it would be convenient to have them all in one place.